### PR TITLE
added globalExeptionHandler and validation for classes dto

### PR DESCRIPTION
**Pull request by Dima23-ops5**

### CODE REVIEW

File: src/main/java/mate/academy/StringBootIntro/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,14 +1,22 @@
 package mate.academy.StringBootIntro.dto;
 
 import java.math.BigDecimal;
+
+import jakarta.validation.constraints.NotNull;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/dto/CreateBookRequestDto.java]: `@NotBlank` for String
Check everywhere

File: src/main/java/mate/academy/StringBootIntro/dto/BookDto.java [hunk 2]
```diff
@@ -1,15 +1,24 @@
 package mate.academy.StringBootIntro.dto;
 
 import java.math.BigDecimal;
+
+import jakarta.validation.constraints.NotNull;
 import lombok.Data;
 
 @Data
 public class BookDto {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/dto/BookDto.java]: Is it response? Don't need to validate it

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's improve your solution

File: src/main/java/mate/academy/StringBootIntro/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
+    @NotBlank
+    @NotNull(message = "Title can`t be null")
     private String title;
+    @NotBlank
+    @NotNull(message = "Author can`t be null")
     private String author;
+    @NotBlank
+    @NotNull(message = "Isbn can`t be null")
     private String isbn;
+    @NotNull(message = "Price can`t be null and also will be greater than 0")
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/StringBootIntro/dto/CreateBookRequestDto.java]: Add @Positive

File: src/main/java/mate/academy/StringBootIntro/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,14 +1,27 @@
...
     private BigDecimal price;
+    @NotBlank
+    @NotNull(message = "Description can`t be null")
     private String description;
+    @NotBlank
+    @NotNull(message = "CoverImage can`t be null")
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/StringBootIntro/dto/CreateBookRequestDto.java]: Try to use more direct annotations, not only @Notblank

File: src/main/java/mate/academy/StringBootIntro/exeption/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getAllErrors().stream()
+                .map(e -> getErrorMessage(e))
+                .toList();
+        body.put("errors", errors);
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/StringBootIntro/exeption/CustomGlobalExceptionHandler.java]: Lets save strings to constants

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Review by sarakhmen — APPROVED** [review]

**Pull request closed without merging**